### PR TITLE
[RFC] vo_gpu: remove deprecated options --gamma-factor and --gamma-auto

### PR DESCRIPTION
**Pull request by Akemi**
--gamma-factor and --gamma-auto were deprecated in 0.35 (ac39661 and 2207236). the latter was only used on macOS.
the equivalent client API functionality wasn't officially deprecated.

what i would like to discuss:
do we want to remove those options already?
if yes, how do we want to remove them? since the client API equivalent wasn't deprecated, removal of the functionality can't be done immediately?
do we want to deprecate the client API functionality first in the next release and remove the functionality in one of the releases after the next one?
the client API deprecation change should probably be taken separately?

hope i found all the dependent code and removed it properly.

### CODE REVIEW

File: video/out/vo.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ enum {
     VO_EVENT_ICC_PROFILE_CHANGED        = 1 << 2,
     // Some other window state changed (position, window state, fps)
     VO_EVENT_WIN_STATE                  = 1 << 3,
-    // The ambient light conditions changed and need to be reloaded
+    // Deprecated. The ambient light conditions changed and need to be reloaded
     VO_EVENT_AMBIENT_LIGHTING_CHANGED   = 1 << 4,
```
Comment (kasper93) [inline video/out/vo.h:44]: This can be removed too, no? It is not public API.

Comment (Akemi) [reply video/out/vo.h:44]: yeah, i wasn't sure how to handle this since we would get a 'hole' in that enum or would we want to renumber it?

Comment (kasper93) [reply video/out/vo.h:44]: hole is fine, I think, to keep it simple. There are plenty of bits left for new values.

**Comment by ruihe774** [comment]
FWIW I wonder what is the alternative of `--gamma-factor`? I am using `--gamma-factor=1.2239` on macOS to match the color with QuickTime Player.

**Comment by hooke007** [comment]
https://github.com/mpv-player/mpv/pull/10699#issuecomment-1325851500 https://github.com/mpv-player/mpv/issues/11153
But it doesn't give the real answer.

**Comment by Dudemanguy** [comment]
I'm not sure `--gamma-auto` has any use but I agree with @na-na-hi that `--gamma-factor` is worth keeping. It's just a simple multiplier anyway so it's not like there's great code complexity.

File: video/out/vo.h (hunk 1 above)
Comment (Dudemanguy) [reply video/out/vo.h:44]: If something new is added later, we can always just fill the hole in too.

**Comment by Akemi** [comment]
why was this option deprecated in the first place?

**Comment by hooke007** [comment]
https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU

`--gamma-factor and --gamma-auto: considered deprecated by design, the new BPC logic handles this much better`

Not sure if it was related.

**Comment by Akemi** [comment]
from the commits and the PR i could only gather "outdated information" and "--gamma-auto not working". though both aren't really a reason to deprecate the options. for the former update the info, for the latter fix it or open an issue (without knowing what exactly is broken, hard to judge).

seems like the initial use case for `--gamma-factor` is gone, though it still has a more general use case of being the only way to adjust gamma at all (besides video filters via ffmpeg i guess?).

then i am also of the opinion to undeprecate both options.

**Comment by ruihe774** [comment]
> https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU
> 
> `--gamma-factor and --gamma-auto: considered deprecated by design, the new BPC logic handles this much better`
> 
> Not sure if it was related.

I wonder what is "BPC"? I cannot find the word in any other places.

**Comment by Akemi** [comment]
black point compensation, is my guess.

**Comment by Akemi** [comment]
closing this in favour of #14540.

**Pull request closed without merging**